python_modules/dagster/dagster/_core/execution/run_metrics_thread.py

```diff
@@ -110,16 +110,12 @@ def _report_run_metrics(
     metrics: Dict[str, float],
     run_tags: Dict[str, str],
 ):
-    datapoints: List[TelemetryDataPoint] = []
+    datapoints: Dict[str, float] = {}
     for metric, value in metrics.items():
         if value is None:
             continue
         try:
-            datapoint = TelemetryDataPoint(
-                name=metric,
-                value=float(value),
-            )
-            datapoints.append(datapoint)
+            datapoints[metric] = float(value)
```

Comment:
Not to prematurely optimize, but are we worried at all about efficiency of repeating the same metric over and over again if there are many values? Did we consider a List[Dict[str, List[float]]] as well? Probably depends on the expected volume

Sending this up over http rather than graphql if we want to gzip could help allay that concern too

Reply:
see global comment